src/features/auth/whoami.ts

```diff
@@ -62,3 +108,98 @@ export const whoami: RequestHandler = async (req, res) => {
 		res.status(500).end();
 	}
 };
+
+export const actorWhoami: RequestHandler = async (req, res) => {
+	try {
+		const actorInfo = await sbvrUtils.db.readTransaction(async (tx) => {
+			const actorId = req.apiKey?.actor ?? req.user?.actor;
+
+			if (actorId == null) {
+				throw new UnauthorizedError(
+					'Request API Key or Token has no associated actor',
+				);
+			}
+
+			return (await api.resin.get({
+				resource: 'actor',
+				passthrough: { req: permissions.root, tx },
```

Comment:
I don't think doing this with root permissions is good if it's an api key associated to a user but which would not be expected to be able to get the username/email

Reply:
Thanks @Page- ! This is a nice catch. I added some logic before it similar to the user whoami endpoint to validate that when this is an user key, the user has the needed permissions before calling resource: 'actor' with root permissions. I also added tests for it